devito/types/dense.py

```diff
@@ -1457,6 +1457,12 @@ class SubFunction(Function):
     DiscreteFunction it's bound to.
     """
 
+    __rkwargs__ = DiscreteFunction.__rkwargs__ + ('parent', 'dimensions', 'shape')
```

Comment:
why are dimensions and shape needed?

Reply:
SubFunction are built without a grid as they only rely on the parent and get the distributor from the parent as well so they need dimensions and shape to rebuild correctly.